lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp

```diff
@@ -283,14 +287,149 @@ struct ConvolutionToConv2dPattern
     return success();
   }
 
-  LogicalResult
-  matchAndRewrite(ttir::ConvolutionOp op, OpAdaptor adaptor,
-                  ConversionPatternRewriter &rewriter) const override {
+  ttir::ReshapeOp createReshapeOp(Location loc, Value tensor,
+                                  llvm::ArrayRef<int64_t> target_input_shape,
+                                  ::mlir::ArrayAttr constraints,
+                                  ConversionPatternRewriter &rewriter) const {
+    auto inputType = mlir::cast<RankedTensorType>(tensor.getType());
+
+    auto output = rewriter.create<tensor::EmptyOp>(
```

Comment:
Rename `output` to something else to signify that it's an op, since the type is `tensor::EmptyOp`, not an output `Value` of the op itself.